src/app/beer_garden/scheduler.py

```diff
@@ -512,9 +520,11 @@ def run_job(job_id, request_template, **kwargs):
             logger.debug(f"{db_job!r} request completed with SUCCESS status")
             updates["inc__success_count"] = 1
 
-        db.modify(db_job, **updates)
+        if updates != {}:
+            db.modify(db_job, **updates)
     except Exception as ex:
         logger.exception(f"Error executing {db_job}: {ex}")
```

Comment:
Not related to your changes, but `logger.exception` isn't intended to be used in this context.  Per the [logging docs](https://docs.python.org/3/library/logging.html#logging.Logger.exception):

>  This method should only be called from an exception handler.

Since you're here, let's change this to `logger.error` to clean this up.